src/charm.py

```diff
@@ -434,15 +480,19 @@ def _on_config_changed(self, _: ConfigChangedEvent):
         # reconsider all data sent over the relations and all configs
         new_external_host = self._external_host
         new_routing_mode = self.config["routing_mode"]
+        new_forward_auth_mode = self._is_forward_auth_enabled
+
         # TODO set BlockedStatus here when compound_status is introduced
         #  https://github.com/canonical/operator/issues/665
 
         if (
             self._stored.current_external_host != new_external_host  # type: ignore
             or self._stored.current_routing_mode != new_routing_mode  # type: ignore
+            or self._stored.current_forward_auth_mode != new_forward_auth_mode  # type: ignore
         ):
             self._stored.current_external_host = new_external_host  # type: ignore
             self._stored.current_routing_mode = new_routing_mode  # type: ignore
+            self._stored.current_forward_auth_mode = new_forward_auth_mode  # type: ignore
             self._process_status_and_configurations()
```

Comment:
**Suggestion**: update stored state only after calling `_process_status_and_configurations`.

We had an [issue](https://github.com/canonical/cos-proxy-operator/pull/89) recently when `relation-set` failed, and upon hook retry the relation update logic wasn't executed because stored state was already updated from before.

Since `_process_status_and_configurations` updates reldata, we could call it every time regardless of the stored state, or before we update stored state (and pass args directly instead of relying on stored state?).